pkg/api/controller.go

```diff
@@ -1626,23 +1626,36 @@ func (c *Controller) validateStorageNamespace(storageNamespace string) error {
 }
 
 func (c *Controller) ensureStorageNamespace(ctx context.Context, storageNamespace string) error {
-	const (
-		dummyKey  = "dummy"
-		dummyData = "this is dummy data - created by lakeFS in order to check accessibility"
-	)
+	dummyKey := fmt.Sprintf("%s/dummy", c.Config.Committed.BlockStoragePrefix)
+	const dummyData = "this is dummy data - created by lakeFS in order to check accessibility"
 
-	obj := block.ObjectPointer{
+	// check if dummy file exists in previous lakeFS versions at the root of the storage namespace
+	objLegacyLocation := block.ObjectPointer{
 		StorageNamespace: storageNamespace,
 		IdentifierType:   block.IdentifierTypeRelative,
-		Identifier:       dummyKey,
+		Identifier:       "dummy",
```

Comment:
Suggest keeping this as a key `dummyLegacyKey` or somesuch.  If you do that you probably don't need the comment at l.1632.

Reply:
Changed implementation please look